scarb/tests/build_cairo_plugin.rs

```diff
@@ -252,3 +257,97 @@ fn compile_cairo_plugin_with_other_target() {
             target `cairo-plugin` cannot be mixed with other targets
         "#});
 }
+
+#[test]
+fn can_emit_plugin_warning() {
+    let temp = TempDir::new().unwrap();
+    let t = temp.child("some");
+    simple_project_with_code(
+        &t,
+        indoc! {r#"
+        use cairo_lang_macro::{ProcMacroResult, TokenStream, attribute_macro, Diagnostic};
+
+        #[attribute_macro]
+        pub fn some_macro(token_stream: TokenStream) -> ProcMacroResult {
+            let _code = token_stream.to_string();
+            let diag = Diagnostic::warn("Some warning from macro.");
+            ProcMacroResult::Leave { diagnostics: vec![diag] }
+        }
+        "#},
+    );
+    let project = temp.child("hello");
+    ProjectBuilder::start()
+        .name("hello")
+        .version("1.0.0")
+        .dep("some", &t)
+        .lib_cairo(indoc! {r#"
+            #[some]
+            fn f() -> felt252 { 12 }
+        "#})
+        .build(&project);
+
+    Scarb::quick_snapbox()
+        .arg("build")
+        // Disable output from Cargo.
+        .env("CARGO_TERM_QUIET", "true")
+        .current_dir(&project)
+        .assert()
+        .success()
+        .stdout_matches(indoc! {r#"
+            [..] Compiling some v1.0.0 ([..]Scarb.toml)
+            [..] Compiling hello v1.0.0 ([..]Scarb.toml)
+            warn: Plugin diagnostic: Some warning from macro.
```

Comment:
> `Plugin diagnostic:`

Where does this come from? Is it neccessary? I don't feel we need it?

Reply:
I don't think we need it at all, but it comes from Cairo compiler. 
All diagnostics from plugins are registered as `SemanticDiagnostics`, which format with such prefix. 

https://github.com/starkware-libs/cairo/blob/a771425155ec5ea740eea5e7ee00ea5870b498a3/crates/cairo-lang-semantic/src/db.rs#L1126

https://github.com/starkware-libs/cairo/blob/a771425155ec5ea740eea5e7ee00ea5870b498a3/crates/cairo-lang-semantic/src/diagnostic.rs#L511

Reply:
meh